### PR TITLE
Generaluploads

### PR DESCRIPTION
**Pull request by FR3D99**

### CODE REVIEW

File: commands/images/sendgeneral.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    // Check if the specified subfolder exists
+    if (fs.existsSync(imageFolder)) {
+      // Handle the 'chooseimage' command by sending a random image from the specified subfolder
+      await sendRandomImage(interaction, imageFolder);
+    } else {
+      // Print the full file path in the error message
+      console.log(`Specified subfolder not found. Attempted path: ${imageFolder}`);
+      // Send a message indicating that the specified subfolder does not exist
+      await interaction.reply(`The specified subfolder "${subfolder}" does not exist. Upload something first pls.`);
```
Comment (DylanSbogar) [inline commands/images/sendgeneral.js:30]: `await interaction.reply({content: `The specified subfolder "${subfolder}" does not exist. Upload something first pls.`, flags: MessagFlags.Ephemeral});`

File: commands/images/sendgeneral.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+      // Create an AttachmentBuilder instance with the random image file
+      const imagePath = path.join(imageFolder, randomImageFile);
+      const imageAttachment = new AttachmentBuilder(imagePath);
+
+      // Send the random image file as a message attachment
+      await interaction.reply({ files: [imageAttachment] });
+    } else {
+      // Send a message indicating that the folder is empty
+      await interaction.reply('No images found in the specified folder.');
```
Comment (DylanSbogar) [inline commands/images/sendgeneral.js:54]: await interaction.reply({content: 'No images found in the specified folder.', flags: MessageFlags.Ephemeral});

File: commands/images/uploadgeneral.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      }
+
+      // Download and save the image locally
+      try {
+        const response = await fetch(imageAttachment.url);
+        const buffer = await response.buffer();
+        fs.writeFileSync(filePath, buffer);
+
+        // Send a confirmation message to the user
+        await interaction.reply(`Image saved :)`);
```
Comment (DylanSbogar) [inline commands/images/uploadgeneral.js:59]: await interaction.reply({content: `Image saved :)`, flags: MessageFlags.Ephemeral});

File: commands/images/uploadgeneral.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      } catch (error) {
+        console.error('Error saving the image:', error);
+        await interaction.reply('An error occurred while saving the image.');
```
Comment (DylanSbogar) [inline commands/images/uploadgeneral.js:62]: await interaction.reply({content: 'An error occurred while saving the image.', flags: MessageFlags.Ephemeral});

File: commands/images/sendgeneral.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+const { SlashCommandBuilder, AttachmentBuilder } = require("discord.js");
```
Comment (DylanSbogar) [inline commands/images/sendgeneral.js:1]: Should just need to add `MessageFlags` here for the below to work

File: commands/images/uploadgeneral.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+const { SlashCommandBuilder } = require("discord.js");
```
Comment (DylanSbogar) [inline commands/images/uploadgeneral.js:1]: Should just need to add `MessageFlags` here for the below to work

File: commands/images/uploadgeneral.js (hunk 4 above)
Comment (FR3D99) [inline commands/images/uploadgeneral.js:62]:
```suggestion
        await interaction.reply({content: 'An error occurred while saving the image.', flags: MessageFlags.Ephemeral});
```

Comment (GitAYYYer) [reply commands/images/uploadgeneral.js:62]: dont listen to this guy

**Pull request merged**